### ISSUE TITLE
Provide Context Menus

### ISSUE THREAD

**Issue by alexsb** [lineupjs/lineupjs#508]
We'd like the ability to use context menus going forward. To do that, we need to develop a way for lineup to trigger context menus, but let an external party (like datavisyn) style and customize them (e.g., remove individual entries, or add new ones based on hooks). 

It would also be good if LineUp and datavisyn used the same library to do context menus; I think lineup is using popper.js for menus, and datavisyn wants to use popper.js for tool tips, but might be using bootstrap for menus?

**Comment by sgratzl** [comment]
easiest and most generic solution is using a contextmenu listener on the root element and check which element was pressed. 

e.g. https://codepen.io/sgratzl/pen/RwZvXrQ?editors=0011

**Comment by alexsb** [comment]
@thinkh any thoughts on whether that would be compatible with our needs?

**Comment by thinkh** [comment]
Yes, that would work. The question whether or not we want a context menu for a single cell.

In case we want to reduce/remove the toolbar from the column header and move some parts into the context menu, the context menu should be already part of the LineUp library.

**Comment by alexsb** [comment]
I'd advocate for removing the tool bar from the column header. It results in a lot of buttons and clutters the view.

**Comment by alexsb** [comment]
What would we have to do to move this forward?

**Comment by alexsb** [comment]
Can we be specific on how this would be implemented? @thinkh could you come up with a proposal please?
